File: backend/app/schemas/players.py
"""
Player Schemas
"""

from datetime import datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field

from .base import BaseSchema


class PlayerBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    team: Optional[str] = Field(None, max_length=100)
    position: Optional[str] = Field(None, max_length=50)
    debut_year: Optional[int] = Field(None, ge=1900, le=2100)
    is_rookie: bool = False
    is_prospect: bool = False
    mlb_id: Optional[int] = None


class PlayerCreate(PlayerBase):
    pass


class PlayerUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    team: Optional[str] = Field(None, max_length=100)
    position: Optional[str] = Field(None, max_length=50)
    debut_year: Optional[int] = Field(None, ge=1900, le=2100)
    is_rookie: Optional[bool] = None
    is_prospect: Optional[bool] = None
    mlb_id: Optional[int] = None


class PlayerResponse(BaseSchema):
    id: UUID
    name: str
    name_normalized: str
    team: Optional[str] = None
    position: Optional[str] = None
    debut_year: Optional[int] = None
    is_rookie: bool
    is_prospect: bool
    mlb_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime


class PlayerInventorySummary(BaseSchema):
    """Player-level inventory summary"""
    player_id: UUID
    player_name: str
    team: Optional[str] = None
    position: Optional[str] = None
    unique_cards: int = 0
    total_cards: int = 0
    auto_count: int = 0
    rookie_count: int = 0
    first_bowman_count: int = 0
    numbered_count: int = 0
    total_cost: Decimal = Decimal("0")
    total_revenue: Decimal = Decimal("0")
    profit: Decimal = Decimal("0")

File: frontend/src/types/players.ts
/**
 * Player Types - Aligned with Backend Schemas
 */

export interface Player {
  id: string;
  name: string;
  name_normalized: string;
  team: string | null;
  position: string | null;
  debut_year: number | null;
  is_rookie: boolean;
  is_prospect: boolean;
  mlb_id: number | null;
  created_at: string;
  updated_at: string;
}

export interface PlayerCreate {
  name: string;
  team?: string | null;
  position?: string | null;
  debut_year?: number | null;
  is_rookie?: boolean;
  is_prospect?: boolean;
  mlb_id?: number | null;
}

export interface PlayerUpdate {
  name?: string;
  team?: string | null;
  position?: string | null;
  debut_year?: number | null;
  is_rookie?: boolean;
  is_prospect?: boolean;
  mlb_id?: number | null;
}

export interface PlayerInventorySummary {
  player_id: string;
  player_name: string;
  team: string | null;
  position: string | null;
  unique_cards: number;
  total_cards: number;
  auto_count: number;
  rookie_count: number;
  first_bowman_count: number;
  numbered_count: number;
  total_cost: number;
  total_revenue: number;
  profit: number;
}
